### PR TITLE
feat(console,phrases): implement application permissions assignment 4-4

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Implement application permissions assignment using the newly added `DataTransferBox` component.

image

image

## Testing

Test locally
Will add integration tests later

## Checklist

- [ ] ~`.changeset`~
- [ ] ~unit tests~
- [ ] ~integration tests~
- [ ] ~necessary TSDoc comments~

### CODE REVIEW

File: packages/console/src/pages/ApplicationDetails/components/Permissions/ApplicationScopesAssignmentModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        <DataTransferBox {...scopesAssignment[ApplicationUserConsentScopeType.ResourceScopes]} />
+      </TabWrapper>
+      <TabWrapper
+        key={ApplicationUserConsentScopeType.OrganizationScopes}
+        isActive={ApplicationUserConsentScopeType.OrganizationScopes === activeTab}
+      >
+        <DataTransferBox
+          {...scopesAssignment[ApplicationUserConsentScopeType.OrganizationScopes]}
+        />
+      </TabWrapper>
```
Comment (xiaoyijun) [inline packages/console/src/pages/ApplicationDetails/components/Permissions/ApplicationScopesAssignmentModal/index.tsx:94]: The code is repetitive, can it be optimized?

Comment (simeng-li) [reply packages/console/src/pages/ApplicationDetails/components/Permissions/ApplicationScopesAssignmentModal/index.tsx:94]: Good question.  These data props types are different. This element will infer the data type from its input properties. As well as the handler's type.  Using a map render here will throw a type not match error.  

e.g data type A is not assignable to the handler type B

**Review by charIeszhao — APPROVED** [review]

File: packages/console/src/pages/ApplicationDetails/components/Permissions/ApplicationScopesAssignmentModal/index.tsx (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/components/Permissions/ApplicationScopesAssignmentModal/index.tsx]: Gotcha~

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Pull request merged**